sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultPlanScheduler.java

```diff
@@ -133,26 +134,43 @@ public DefaultPlanScheduler(
     private void killTasks(PodInstanceRequirement podInstanceRequirement) {
         Map<String, TaskInfo> taskInfoMap = new HashMap<>();
         stateStore.fetchTasks().forEach(taskInfo -> taskInfoMap.put(taskInfo.getName(), taskInfo));
-
-        List<String> taskNames = TaskUtils.getTaskNames(
-                podInstanceRequirement.getPodInstance(),
+        logger.info("Killing tasks for pod instance requirement: {}:{}",
+                podInstanceRequirement.getPodInstance().getName(),
                 podInstanceRequirement.getTasksToLaunch());
 
-        taskNames = taskNames.stream()
-                .filter(taskName -> taskInfoMap.containsKey(taskName))
+        List<TaskSpec> taskSpecsToLaunch = podInstanceRequirement.getPodInstance().getPod().getTasks().stream()
+                .filter(taskSpec -> podInstanceRequirement.getTasksToLaunch().contains(taskSpec.getName()))
+                .collect(Collectors.toList());
+        logger.info("TaskSpecs to launch: {}",
+                taskSpecsToLaunch.stream().map(taskSpec -> taskSpec.getName()).collect(Collectors.toList()));
+
+        List<String> resourceSetsToConsume = taskSpecsToLaunch.stream()
+                .map(taskSpec -> taskSpec.getResourceSet())
+                .map(resourceSet -> resourceSet.getId())
+                .collect(Collectors.toList());
+        logger.info("Resource sets to consume: {}",
+                podInstanceRequirement.getPodInstance().getName(),
+                resourceSetsToConsume);
+
+        List<String> tasksToKill = podInstanceRequirement.getPodInstance().getPod().getTasks().stream()
+                .filter(taskSpec -> resourceSetsToConsume.contains(taskSpec.getResourceSet().getId()))
+                .map(taskSpec -> TaskSpec.getInstanceName(podInstanceRequirement.getPodInstance(), taskSpec))
                 .collect(Collectors.toList());
+        logger.info("Tasks to kill: {}", tasksToKill);
 
-        for (String taskName : taskNames) {
+        for (String taskName : tasksToKill) {
             TaskInfo taskInfo = taskInfoMap.get(taskName);
-            Optional<Protos.TaskStatus> taskStatusOptional = stateStore.fetchStatus(taskInfo.getName());
+            if (taskInfo != null) {
```

Comment:
I wanted the logging.  I wanted to see what the system wanted to kill, pre-filtered.